packages/wrangler/src/init.ts

```diff
@@ -942,25 +929,33 @@ async function getWorkerConfig(
 			(durableObject) => (durableObject as { class_name: string }).class_name
 		);
 
-	const routeOrRoutes = routes.map((rawRoute) => {
-		const { id: _id, ...route } = rawRoute;
-		if (Object.keys(route).length === 1) {
-			return route.pattern;
-		} else {
-			return route as Route;
-		}
-	});
-	const routeOrRoutesToConfig =
-		routeOrRoutes.length > 1
-			? { routes: routeOrRoutes }
-			: { route: routeOrRoutes[0] };
+	const allRoutes: Route[] = [
+		...routes.map(
+			(r) => ({ pattern: r.pattern, zone_name: r.zone_name } as ZoneNameRoute)
+		),
+		...customDomains.map(
+			(c) =>
+				({
+					pattern: c.hostname,
+					zone_name: c.zone_name,
+					custom_domain: true,
+				} as CustomDomainRoute)
+		),
+	];
 
 	return {
+		name: workerName,
+		main: entrypoint,
+		workers_dev: workersDev.enabled,
 		compatibility_date:
 			serviceEnvMetadata.script.compatibility_date ??
 			new Date().toISOString().substring(0, 10),
-		...routeOrRoutesToConfig,
-		usage_model: serviceEnvMetadata.script.usage_model,
+		compatibility_flags: serviceEnvMetadata.script.compatibility_flags,
+		...(allRoutes.length ? { routes: allRoutes } : {}),
+		usage_model:
```

Comment:
We just noticed this ourselves and we decided that we wanted `usage_model` to be unspecified if the user has standard instead of just the Worker using the standard usage model. I implemented this in [this commit](https://github.com/zebp/workers-sdk/commit/3881974852f56cc9c8a44b458527010fa9a1cf9f#diff-fa9a52007c2b6571335f75c75e00cf783306b826a082fe144ac0337b3bfff9fcR969) before realizing you had this PR up.